tor/onionfile/onion_file_test.go

```diff
@@ -0,0 +1,95 @@
+package onionfile
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	privateKey = []byte("RSA1024 hide_me_plz")
+)
+
+type MockEncrypter struct{}
+
+func (m MockEncrypter) EncryptPayloadToWriter(_ bytes.Buffer, _ io.Writer,
+	_ []byte) error {
+
+	return nil
+}
+
+func (m MockEncrypter) DecryptPayloadFromReader(_ io.Reader, _ []byte) ([]byte,
+	error) {
+
+	return privateKey, nil
+}
+
+// TestOnionFile tests that the File implementation of the OnionStore
+// interface behaves as expected.
+func TestOnionFile(t *testing.T) {
+	t.Parallel()
+
+	tempDir, err := ioutil.TempDir("", "onion_store")
+	require.NoError(t, err, "unable to create temp dir")
+
+	privateKeyPath := filepath.Join(tempDir, "secret")
+
+	mockEncrypter := MockEncrypter{}
+
+	// Create a new file-based onion store. A private key should not exist
+	// yet.
+	onionFile := NewOnionFile(
+		privateKeyPath, 0600, false, mockEncrypter,
+		make([]byte, 0),
+	)
+	if _, err := onionFile.PrivateKey(V2); err != ErrNoPrivateKey {
```

Comment:
Use `require.ErrorIs()` here and `require` everywhere else instead of `t.Fatalf()`.